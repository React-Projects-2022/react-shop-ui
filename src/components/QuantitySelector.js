import React, { useEffect } from "react";
import PropTypes from "prop-types";

import "./../styles/quantity-selector.css";
import { useCounter } from "../hooks/useCounter";

export const QuantitySelector = ({ stock, updateValue }) => {
  const { counter, increment, decrement } = useCounter(1);

  const changeValue = (action) => {
    if (action === "-" && counter > 1) {
      decrement();
    } else if (action === "+" && counter < stock) {
      increment();
    }
  };

  useEffect(() => {
    updateValue(counter);
  }, [updateValue, counter]);

  return (
    <>
      <div>
        <button
          type="button"
          className="btn btn-minus qty-btn"
          onClick={() => changeValue("-")}
        >
          -
        </button>
        <input
          type="text"
          value={counter}
          name="quantity"
          className="text-center quantity"
          disabled
        />
        <button
          type="button"
          className="btn btn-plus qty-btn"
          onClick={() => changeValue("+")}
        >
          +
        </button>
      </div>
      {counter === stock ? (
        <p className="alert alert-danger">
          No puedes seleccionar m√°s cantidad por no haber + stock
        </p>
      ) : (
        ""
      )}
    </>
  );
};

QuantitySelector.propTypes = {
  stock: PropTypes.number.isRequired,
  updateValue: PropTypes.func.isRequired,
};
