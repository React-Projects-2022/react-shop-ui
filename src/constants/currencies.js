export const CURRENCY_LIST = {
  EURO: "EUR",
  US_DOLLAR: "USD",
  BRITISH_POUND: "GBP",
  CANADIAN_DOLAR: "CAD",
  COLOMBIAN_PESO: "COP",
  CHILEAN_PESO: "CLP",
  MEXICAN_PESO: "MXN",
  VENEZUELA_BOLIVAR: "VSD",
  PERUVIAN_SOL: "PEN",
};

export const CURRENCIES_SYMBOL = {
  EUR: "€",
  USD: "$",
  GBP: "£",
  CAD: "$",
  COP: "$",
  CLP: "$",
  MXN: "$",
  VSD: "Bs.",
  PEN: "S/.",
};
