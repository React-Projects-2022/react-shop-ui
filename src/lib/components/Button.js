import React from "react";
import PropTypes from "prop-types";
const Button = ({ text, className, handleAction, params }) => (
  <button
    type="button"
    className={className}
    onClick={() => handleAction(params)}
  >
    {text}
  </button>
);

export { Button };

Button.propTypes = {
  handleAction: PropTypes.func.isRequired,
};
