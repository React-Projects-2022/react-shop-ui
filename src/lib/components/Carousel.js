import React from "react";
import PropTypes, { object } from "prop-types";
import "./../styles/carousel.css";
const Carousel = ({ carousel }) => (
  <header>
    <div
      id="carouselExampleCaptions"
      className="carousel slide"
      data-bs-ride="carousel"
      data-bs-interval="3000"
    >
      <div className="carousel-indicators">
        {carousel.map((item, index) => (
          <button
            key={item.id ? item.id : index}
            type="button"
            data-bs-target="#carouselExampleCaptions"
            data-bs-slide-to={index}
            className={index === 0 ? "active" : ""}
            aria-current="true"
            aria-label={"Slide ".concat(index + 1)}
          ></button>
        ))}
      </div>
      <div className="carousel-inner">
        {carousel.map((item, index) => (
          <div
            key={item.id ? item.id : index}
            className={index === 0 ? "carousel-item active" : "carousel-item"}
            style={{
              backgroundImage: `url("${item.background}")`,
            }}
          >
            <div className="carousel-caption">
              <h5>{item.title}</h5>
              <p>{item.description}</p>
            </div>
          </div>
        ))}
      </div>
      <button
        className="carousel-control-prev"
        type="button"
        data-bs-target="#carouselExampleCaptions"
        data-bs-slide="prev"
      >
        <span className="carousel-control-prev-icon" aria-hidden="true"></span>
        <span className="visually-hidden">Previous</span>
      </button>
      <button
        className="carousel-control-next"
        type="button"
        data-bs-target="#carouselExampleCaptions"
        data-bs-slide="next"
      >
        <span className="carousel-control-next-icon" aria-hidden="true"></span>
        <span className="visually-hidden">Next</span>
      </button>
    </div>
  </header>
);

export { Carousel };

Carousel.propTypes = {
  carousel: PropTypes.array.isRequired
};


/**
 * {
          id: item.id,
          title: item.product.name,
          description: item.platform.name,
          url: window.location.origin.concat(`/games/details/${item.id}`),
          background: item.product.img,
        }
 */