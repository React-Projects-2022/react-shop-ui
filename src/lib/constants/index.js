import { CURRENCIES_SYMBOL, CURRENCY_LIST } from "./currencies";
const CONSTANTS = { CURRENCIES_SYMBOL, CURRENCY_LIST }
export { CONSTANTS };