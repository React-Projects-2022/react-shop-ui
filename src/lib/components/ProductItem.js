import "./../styles/product-item.css";
import { Rating } from "./Rating";
import PropTypes from "prop-types";
import { Button } from "./Button";
import { CONSTANTS } from "../constants";
export const ProducItem = ({
  item,
  showDetails,
  addCart,
  showDesc,
  selectMoney,
}) => {
  // https://bbbootstrap.com/snippets/product-card-template-hover-effect-and-animation-23224168
  return (
    <>
      {item ? (
        <div className="card">
          <div className="heart">
            <i className="fa fa-heart"></i>
          </div>
          <div className="top-div">
            <div className="border">
              <img src={item.product.img} alt="" />
            </div>
            <span>
              {CONSTANTS.CURRENCIES_SYMBOL[selectMoney]} {item.price}
            </span>
          </div>
          <div className="bottom-div">
            <h5 className="truncate-one-line">{item.product.name}</h5>
            {showDesc ? <p>{item.platform.name}</p> : <br />}
            <Rating
              max={5}
              ratingItem={{
                value: item.product.rating.value,
                count: item.product.rating.count,
              }}
            />
          </div>
          <div className="last-section text-center">
            <div className="buttons">
              <Button
                params={item.id}
                text={<i className="fas fa-info-circle"></i>}
                handleAction={showDetails}
              />
              &nbsp;
              <Button
                params={item.id}
                text={<i className="fas fa-cart-plus"></i>}
                handleAction={addCart}
              />
            </div>
          </div>
        </div>
      ) : (
        ""
      )}
    </>
  );
};

ProducItem.propTypes = {
  item: PropTypes.object.isRequired,
  showDetails: PropTypes.func.isRequired,
  addCart: PropTypes.func.isRequired,
  selectMoney: PropTypes.oneOf(Object.values(CONSTANTS.CURRENCY_LIST)),
};
