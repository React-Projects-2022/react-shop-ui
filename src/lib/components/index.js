import React from "react";
import { QuantitySelector } from "./components/QuantitySelector";
import { ProductItem } from "./components/ProductItem";
import { ProductItemGroup } from "./components/ProductItemGroup";
import { Rating } from "./components/Rating";
import { Button } from "./components/Button";
import { Carousel } from "./components/Carousel";
import { Cart } from "./components/Cart";

export {
  Button,
  Carousel,
  Cart,
  ProductItem,
  ProductItemGroup,
  Rating,
  QuantitySelector,
};
