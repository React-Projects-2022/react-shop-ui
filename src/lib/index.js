import { QuantitySelector } from "./components/QuantitySelector";
import { CONSTANTS } from "./constants";
import { ProducItem } from "./components/ProductItem";
import { ProductItemGroup } from "./components/ProductItemGroup";
import { Rating } from "./components/Rating";
import { Button } from "./components/Button";
import { Carousel } from "./components/Carousel";
export {
  Button,
  Carousel,
  ProducItem,
  ProductItemGroup,
  Rating,
  QuantitySelector,
  CONSTANTS,
};
