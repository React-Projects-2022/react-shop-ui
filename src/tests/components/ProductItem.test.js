import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { ProductItem } from "./../../lib/components/ProductItem";
import { CONSTANTS } from "../../lib/constants";
describe("Pruebas en ProductItem", () => {
  let wrapper;
  const text = `<i className="fas fa-info-circle"></i>`;
  const handleAction = jest.fn();
  const showDetails = jest.fn();
  const addCart = jest.fn();
  const selectMoney = CONSTANTS.CURRENCY_LIST.EURO;
  const showDesc = true;
  const item = {
    id: "1222",
    price: 23,
    product: {
      img: "",
      name: "Super Mario",
      rating: {
        value: 1,
        count: 340,
      },
    },
    platform: {
      name: "Nintendo 64",
    },
  };
  beforeEach(() => {
    wrapper = shallow(
      <ProductItem
        item={item}
        addCart={addCart}
        showDesc={showDesc}
        showDetails={showDetails}
        selectMoney={selectMoney}
      />
    );
  });
  test("Debe de mostrar <ProductItem /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Comprobar que tenemos dos botones y Rating", () => {
    expect(wrapper.find("Button").length).toBe(2);
    expect(wrapper.find("Rating").length).toBe(1);
  });

  test("Comprobar las propiedades de los botones", () => {
    const buttonComponent = wrapper.find("Button");
    expect(buttonComponent.at(0).prop("params")).toBe(item.id);
    expect(buttonComponent.at(1).prop("params")).toBe(item.id);
    expect(buttonComponent.at(0).prop("text")).toMatchObject(
      <i className="fas fa-info-circle"></i>
    );
    expect(buttonComponent.at(1).prop("text")).toMatchObject(
      <i className="fas fa-cart-plus"></i>
    );
    // expect(buttonComponent.at(0).prop("handleAction")).toBeDefined(showDetails);
    // expect(buttonComponent.at(1).prop("handleAction")).toBeDefined(addCart);
  });

  test("Comprobar elementos con los datos si son correctos o no", () => {
    const productNameContent = wrapper.find(".truncate-one-line").text();
    expect(productNameContent).toBe(item.product.name);

    const productDescriptionElement = wrapper.find(".description");
    showDesc
      ? expect(productDescriptionElement.text()).toBe(item.platform.name)
      : expect(productDescriptionElement).toMatchObject({});

    const priceContent = wrapper.find(".top-div span").text();
    expect(priceContent).toBe(`â‚¬ ${item.price}`);
  });

  test("Comprobar que seleccionamos correctamente las monedas para mostrar", () => {
    const currenciesValues = Object.values(CONSTANTS.CURRENCY_LIST);
    Object.keys(CONSTANTS.CURRENCY_LIST).map((__, index) => {
      checkIfMoneySymbolIsCorrect(
        currenciesValues[index],
        CONSTANTS.CURRENCIES_SYMBOL[currenciesValues[index]],
        item
      );
    });
  });
});

const checkIfMoneySymbolIsCorrect = (selectMoney, expectedSymbol, item) => {
  const product = shallow(
    <ProductItem
      item={item}
      addCart={jest.fn()}
      showDesc={false}
      showDetails={jest.fn()}
      selectMoney={selectMoney}
    />
  );
  const priceContent = product.find(".top-div span").text();
  expect(priceContent).toBe(`${expectedSymbol} ${item.price}`);

  expect(product).toMatchSnapshot();
};
