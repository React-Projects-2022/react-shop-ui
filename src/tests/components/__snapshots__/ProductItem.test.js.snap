// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductItem Comprobar que seleccionamos correctamente las monedas para mostrar 1`] = `
<Fragment>
  <div
    className="card"
  >
    <div
      className="heart"
    >
      <i
        className="fa fa-heart"
      />
    </div>
    <div
      className="top-div"
    >
      <div
        className="border"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <span>
        €
         
        23
      </span>
    </div>
    <div
      className="bottom-div"
    >
      <h5
        className="truncate-one-line"
      >
        Super Mario
      </h5>
      <br />
      <Rating
        max={5}
        ratingItem={
          Object {
            "count": 340,
            "value": 1,
          }
        }
      />
    </div>
    <div
      className="last-section text-center"
    >
      <div
        className="buttons"
      >
        <Button
          handleAction={[MockFunction]}
          params="1222"
          text={
            <i
              className="fas fa-info-circle"
            />
          }
        />
         
        <Button
          handleAction={[MockFunction]}
          params={
            Object {
              "id": "1222",
              "platform": Object {
                "name": "Nintendo 64",
              },
              "price": 23,
              "product": Object {
                "img": "",
                "name": "Super Mario",
                "rating": Object {
                  "count": 340,
                  "value": 1,
                },
              },
            }
          }
          text={
            <i
              className="fas fa-cart-plus"
            />
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Pruebas en ProductItem Comprobar que seleccionamos correctamente las monedas para mostrar 2`] = `
<Fragment>
  <div
    className="card"
  >
    <div
      className="heart"
    >
      <i
        className="fa fa-heart"
      />
    </div>
    <div
      className="top-div"
    >
      <div
        className="border"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <span>
        $
         
        23
      </span>
    </div>
    <div
      className="bottom-div"
    >
      <h5
        className="truncate-one-line"
      >
        Super Mario
      </h5>
      <br />
      <Rating
        max={5}
        ratingItem={
          Object {
            "count": 340,
            "value": 1,
          }
        }
      />
    </div>
    <div
      className="last-section text-center"
    >
      <div
        className="buttons"
      >
        <Button
          handleAction={[MockFunction]}
          params="1222"
          text={
            <i
              className="fas fa-info-circle"
            />
          }
        />
         
        <Button
          handleAction={[MockFunction]}
          params={
            Object {
              "id": "1222",
              "platform": Object {
                "name": "Nintendo 64",
              },
              "price": 23,
              "product": Object {
                "img": "",
                "name": "Super Mario",
                "rating": Object {
                  "count": 340,
                  "value": 1,
                },
              },
            }
          }
          text={
            <i
              className="fas fa-cart-plus"
            />
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Pruebas en ProductItem Comprobar que seleccionamos correctamente las monedas para mostrar 3`] = `
<Fragment>
  <div
    className="card"
  >
    <div
      className="heart"
    >
      <i
        className="fa fa-heart"
      />
    </div>
    <div
      className="top-div"
    >
      <div
        className="border"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <span>
        £
         
        23
      </span>
    </div>
    <div
      className="bottom-div"
    >
      <h5
        className="truncate-one-line"
      >
        Super Mario
      </h5>
      <br />
      <Rating
        max={5}
        ratingItem={
          Object {
            "count": 340,
            "value": 1,
          }
        }
      />
    </div>
    <div
      className="last-section text-center"
    >
      <div
        className="buttons"
      >
        <Button
          handleAction={[MockFunction]}
          params="1222"
          text={
            <i
              className="fas fa-info-circle"
            />
          }
        />
         
        <Button
          handleAction={[MockFunction]}
          params={
            Object {
              "id": "1222",
              "platform": Object {
                "name": "Nintendo 64",
              },
              "price": 23,
              "product": Object {
                "img": "",
                "name": "Super Mario",
                "rating": Object {
                  "count": 340,
                  "value": 1,
                },
              },
            }
          }
          text={
            <i
              className="fas fa-cart-plus"
            />
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Pruebas en ProductItem Comprobar que seleccionamos correctamente las monedas para mostrar 4`] = `
<Fragment>
  <div
    className="card"
  >
    <div
      className="heart"
    >
      <i
        className="fa fa-heart"
      />
    </div>
    <div
      className="top-div"
    >
      <div
        className="border"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <span>
        $
         
        23
      </span>
    </div>
    <div
      className="bottom-div"
    >
      <h5
        className="truncate-one-line"
      >
        Super Mario
      </h5>
      <br />
      <Rating
        max={5}
        ratingItem={
          Object {
            "count": 340,
            "value": 1,
          }
        }
      />
    </div>
    <div
      className="last-section text-center"
    >
      <div
        className="buttons"
      >
        <Button
          handleAction={[MockFunction]}
          params="1222"
          text={
            <i
              className="fas fa-info-circle"
            />
          }
        />
         
        <Button
          handleAction={[MockFunction]}
          params={
            Object {
              "id": "1222",
              "platform": Object {
                "name": "Nintendo 64",
              },
              "price": 23,
              "product": Object {
                "img": "",
                "name": "Super Mario",
                "rating": Object {
                  "count": 340,
                  "value": 1,
                },
              },
            }
          }
          text={
            <i
              className="fas fa-cart-plus"
            />
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Pruebas en ProductItem Comprobar que seleccionamos correctamente las monedas para mostrar 5`] = `
<Fragment>
  <div
    className="card"
  >
    <div
      className="heart"
    >
      <i
        className="fa fa-heart"
      />
    </div>
    <div
      className="top-div"
    >
      <div
        className="border"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <span>
        $
         
        23
      </span>
    </div>
    <div
      className="bottom-div"
    >
      <h5
        className="truncate-one-line"
      >
        Super Mario
      </h5>
      <br />
      <Rating
        max={5}
        ratingItem={
          Object {
            "count": 340,
            "value": 1,
          }
        }
      />
    </div>
    <div
      className="last-section text-center"
    >
      <div
        className="buttons"
      >
        <Button
          handleAction={[MockFunction]}
          params="1222"
          text={
            <i
              className="fas fa-info-circle"
            />
          }
        />
         
        <Button
          handleAction={[MockFunction]}
          params={
            Object {
              "id": "1222",
              "platform": Object {
                "name": "Nintendo 64",
              },
              "price": 23,
              "product": Object {
                "img": "",
                "name": "Super Mario",
                "rating": Object {
                  "count": 340,
                  "value": 1,
                },
              },
            }
          }
          text={
            <i
              className="fas fa-cart-plus"
            />
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Pruebas en ProductItem Comprobar que seleccionamos correctamente las monedas para mostrar 6`] = `
<Fragment>
  <div
    className="card"
  >
    <div
      className="heart"
    >
      <i
        className="fa fa-heart"
      />
    </div>
    <div
      className="top-div"
    >
      <div
        className="border"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <span>
        $
         
        23
      </span>
    </div>
    <div
      className="bottom-div"
    >
      <h5
        className="truncate-one-line"
      >
        Super Mario
      </h5>
      <br />
      <Rating
        max={5}
        ratingItem={
          Object {
            "count": 340,
            "value": 1,
          }
        }
      />
    </div>
    <div
      className="last-section text-center"
    >
      <div
        className="buttons"
      >
        <Button
          handleAction={[MockFunction]}
          params="1222"
          text={
            <i
              className="fas fa-info-circle"
            />
          }
        />
         
        <Button
          handleAction={[MockFunction]}
          params={
            Object {
              "id": "1222",
              "platform": Object {
                "name": "Nintendo 64",
              },
              "price": 23,
              "product": Object {
                "img": "",
                "name": "Super Mario",
                "rating": Object {
                  "count": 340,
                  "value": 1,
                },
              },
            }
          }
          text={
            <i
              className="fas fa-cart-plus"
            />
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Pruebas en ProductItem Comprobar que seleccionamos correctamente las monedas para mostrar 7`] = `
<Fragment>
  <div
    className="card"
  >
    <div
      className="heart"
    >
      <i
        className="fa fa-heart"
      />
    </div>
    <div
      className="top-div"
    >
      <div
        className="border"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <span>
        $
         
        23
      </span>
    </div>
    <div
      className="bottom-div"
    >
      <h5
        className="truncate-one-line"
      >
        Super Mario
      </h5>
      <br />
      <Rating
        max={5}
        ratingItem={
          Object {
            "count": 340,
            "value": 1,
          }
        }
      />
    </div>
    <div
      className="last-section text-center"
    >
      <div
        className="buttons"
      >
        <Button
          handleAction={[MockFunction]}
          params="1222"
          text={
            <i
              className="fas fa-info-circle"
            />
          }
        />
         
        <Button
          handleAction={[MockFunction]}
          params={
            Object {
              "id": "1222",
              "platform": Object {
                "name": "Nintendo 64",
              },
              "price": 23,
              "product": Object {
                "img": "",
                "name": "Super Mario",
                "rating": Object {
                  "count": 340,
                  "value": 1,
                },
              },
            }
          }
          text={
            <i
              className="fas fa-cart-plus"
            />
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Pruebas en ProductItem Comprobar que seleccionamos correctamente las monedas para mostrar 8`] = `
<Fragment>
  <div
    className="card"
  >
    <div
      className="heart"
    >
      <i
        className="fa fa-heart"
      />
    </div>
    <div
      className="top-div"
    >
      <div
        className="border"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <span>
        Bs.
         
        23
      </span>
    </div>
    <div
      className="bottom-div"
    >
      <h5
        className="truncate-one-line"
      >
        Super Mario
      </h5>
      <br />
      <Rating
        max={5}
        ratingItem={
          Object {
            "count": 340,
            "value": 1,
          }
        }
      />
    </div>
    <div
      className="last-section text-center"
    >
      <div
        className="buttons"
      >
        <Button
          handleAction={[MockFunction]}
          params="1222"
          text={
            <i
              className="fas fa-info-circle"
            />
          }
        />
         
        <Button
          handleAction={[MockFunction]}
          params={
            Object {
              "id": "1222",
              "platform": Object {
                "name": "Nintendo 64",
              },
              "price": 23,
              "product": Object {
                "img": "",
                "name": "Super Mario",
                "rating": Object {
                  "count": 340,
                  "value": 1,
                },
              },
            }
          }
          text={
            <i
              className="fas fa-cart-plus"
            />
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Pruebas en ProductItem Comprobar que seleccionamos correctamente las monedas para mostrar 9`] = `
<Fragment>
  <div
    className="card"
  >
    <div
      className="heart"
    >
      <i
        className="fa fa-heart"
      />
    </div>
    <div
      className="top-div"
    >
      <div
        className="border"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <span>
        S/.
         
        23
      </span>
    </div>
    <div
      className="bottom-div"
    >
      <h5
        className="truncate-one-line"
      >
        Super Mario
      </h5>
      <br />
      <Rating
        max={5}
        ratingItem={
          Object {
            "count": 340,
            "value": 1,
          }
        }
      />
    </div>
    <div
      className="last-section text-center"
    >
      <div
        className="buttons"
      >
        <Button
          handleAction={[MockFunction]}
          params="1222"
          text={
            <i
              className="fas fa-info-circle"
            />
          }
        />
         
        <Button
          handleAction={[MockFunction]}
          params={
            Object {
              "id": "1222",
              "platform": Object {
                "name": "Nintendo 64",
              },
              "price": 23,
              "product": Object {
                "img": "",
                "name": "Super Mario",
                "rating": Object {
                  "count": 340,
                  "value": 1,
                },
              },
            }
          }
          text={
            <i
              className="fas fa-cart-plus"
            />
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Pruebas en ProductItem Debe de mostrar <ProductItem /> correctamente 1`] = `
<Fragment>
  <div
    className="card"
  >
    <div
      className="heart"
    >
      <i
        className="fa fa-heart"
      />
    </div>
    <div
      className="top-div"
    >
      <div
        className="border"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <span>
        €
         
        23
      </span>
    </div>
    <div
      className="bottom-div"
    >
      <h5
        className="truncate-one-line"
      >
        Super Mario
      </h5>
      <p
        className="description"
      >
        Nintendo 64
      </p>
      <Rating
        max={5}
        ratingItem={
          Object {
            "count": 340,
            "value": 1,
          }
        }
      />
    </div>
    <div
      className="last-section text-center"
    >
      <div
        className="buttons"
      >
        <Button
          handleAction={[MockFunction]}
          params="1222"
          text={
            <i
              className="fas fa-info-circle"
            />
          }
        />
         
        <Button
          handleAction={[MockFunction]}
          params={
            Object {
              "id": "1222",
              "platform": Object {
                "name": "Nintendo 64",
              },
              "price": 23,
              "product": Object {
                "img": "",
                "name": "Super Mario",
                "rating": Object {
                  "count": 340,
                  "value": 1,
                },
              },
            }
          }
          text={
            <i
              className="fas fa-cart-plus"
            />
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;
